libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationLocationProviderTest.java

```diff
@@ -0,0 +1,62 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.location.Location;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+
+import org.junit.Test;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class NavigationLocationProviderTest {
+
+  @Test
+  public void onInitialization_engineListenerIsSet() {
+    LocationEngine locationEngine = mock(LocationEngine.class);
+    NavigationLocationEngineListener listener = mock(NavigationLocationEngineListener.class);
+
+    new NavigationLocationProvider(locationEngine, listener);
+
+    verify(locationEngine).addLocationEngineListener(listener);
+  }
+
+  @Test
+  public void updateLocationEngine_engineListenerIsAdded() {
+    LocationEngine locationEngine = mock(LocationEngine.class);
+    NavigationLocationEngineListener listener = mock(NavigationLocationEngineListener.class);
+    NavigationLocationProvider provider = new NavigationLocationProvider(locationEngine, listener);
+    LocationEngine newLocationEngine = mock(LocationEngine.class);
+
+    provider.updateLocationEngine(newLocationEngine);
+
+    verify(newLocationEngine).addLocationEngineListener(listener);
```

Comment:
I guess we could add the `eq` `ArgumentMatchers` to extra assert that the listener used is the same one that goes through the `addLocationEngineListener`.